google/cloud/security/common/data_access/sql_queries/create_tables.py

```diff
@@ -14,75 +14,207 @@
-CREATE_PROJECT_IAM_POLICIES_TABLE = """
+CREATE_BUCKETS_ACL_TABLE = """
     CREATE TABLE `{0}` (
         `id` bigint(20) unsigned NOT NULL AUTO_INCREMENT,
-        `project_number` bigint(20) DEFAULT NULL,
+        `bucket` varchar(255) DEFAULT NULL,
+        `domain` varchar(255) DEFAULT NULL,
+        `email` varchar(255) DEFAULT NULL,
+        `entity` varchar(255) DEFAULT NULL,
+        `entity_id` varchar(255) DEFAULT NULL,
+        `acl_id` varchar(255) DEFAULT NULL,
+        `kind` varchar(255) DEFAULT NULL,
+        `project_team` json DEFAULT NULL,
         `role` varchar(255) DEFAULT NULL,
-        `member_type` varchar(255) DEFAULT NULL,
-        `member_name` varchar(255) DEFAULT NULL,
-        `member_domain` varchar(255) DEFAULT NULL,
+        `bucket_acl_selflink` varchar(255) DEFAULT NULL,
+        `raw_bucket_acl` json DEFAULT NULL,
         PRIMARY KEY (`id`)
     ) ENGINE=InnoDB DEFAULT CHARSET=utf8;
 """
 
-CREATE_RAW_PROJECT_IAM_POLICIES_TABLE = """
+
+CREATE_BUCKETS_ACL_VIOLATIONS_TABLE = """
     CREATE TABLE `{0}` (
         `id` bigint(20) unsigned NOT NULL AUTO_INCREMENT,
-        `project_number` bigint(20) DEFAULT NULL,
-        `iam_policy` json DEFAULT NULL,
+        `resource_type` varchar(255) NOT NULL,
+        `resource_id` varchar(255) NOT NULL,
+        `rule_name` varchar(255) DEFAULT NULL,
+        `rule_index` int DEFAULT NULL,
+        `violation_type` enum('BUCKET_VIOLATION') NOT NULL,
+        `role` varchar(255) DEFAULT NULL,
+        `entity` varchar(255) DEFAULT NULL,
+        `email` varchar(255) DEFAULT NULL,
+        `domain` varchar(255) DEFAULT NULL,
+        `bucket` varchar(255) DEFAULT NULL,
         PRIMARY KEY (`id`)
     ) ENGINE=InnoDB DEFAULT CHARSET=utf8;
 """
 
-CREATE_ORGANIZATIONS_TABLE = """
+CREATE_BUCKETS_TABLE = """
     CREATE TABLE `{0}` (
-        `org_id` bigint(20) unsigned NOT NULL,
-        `name` varchar(255) NOT NULL,
-        `display_name` varchar(255) DEFAULT NULL,
-        `lifecycle_state` enum('LIFECYCLE_STATE_UNSPECIFIED','ACTIVE',
-            'DELETE_REQUESTED', 'DELETED') NOT NULL,
-        `raw_org` json DEFAULT NULL,
-        `creation_time` datetime DEFAULT NULL,
-        PRIMARY KEY (`org_id`)
+        `id` bigint(20) unsigned NOT NULL AUTO_INCREMENT,
+        `project_number` bigint(20) NOT NULL,
+        `bucket_id` varchar(255) DEFAULT NULL,
+        `bucket_name` varchar(255) DEFAULT NULL,
+        `bucket_kind` varchar(255) DEFAULT NULL,
+        `bucket_storage_class` varchar(255) DEFAULT NULL,
+        `bucket_location` varchar(255) DEFAULT NULL,
+        `bucket_create_time` datetime DEFAULT NULL,
+        `bucket_update_time` datetime DEFAULT NULL,
+        `bucket_selflink` varchar(255) DEFAULT NULL,
+        `bucket_lifecycle_raw` json DEFAULT NULL,
+        `raw_bucket` json DEFAULT NULL,
+        PRIMARY KEY (`id`)
     ) ENGINE=InnoDB DEFAULT CHARSET=utf8;
 """
 
-CREATE_ORG_IAM_POLICIES_TABLE = """
+CREATE_CLOUDSQL_ACL_VIOLATIONS_TABLE = """
```

Comment:
Just an additional note, we might want to discuss later whether to consolidate these multiple violations tables into one table, and still maintain all the needed data that could vary across different resources.  This is good for now.